src/engine/source/indexerconnector/src/indexerConnector.cpp

```diff
@@ -93,14 +93,26 @@ static void builderBulkDelete(std::string& bulkData, std::string_view id, std::s
 
 static void builderBulkIndex(std::string& bulkData, std::string_view id, std::string_view index, std::string_view data)
 {
-    bulkData.append(R"({"index":{"_index":")");
-    bulkData.append(index);
-    bulkData.append(R"(","_id":")");
-    bulkData.append(id);
-    bulkData.append(R"("}})");
-    bulkData.append("\n");
-    bulkData.append(data);
-    bulkData.append("\n");
+    if (!id.empty())
+    {
+        bulkData.append(R"({"index":{"_index":")");
+        bulkData.append(index);
+        bulkData.append(R"(","_id":")");
+        bulkData.append(id);
+        bulkData.append(R"("}})");
+        bulkData.append("\n");
+        bulkData.append(data);
+        bulkData.append("\n");
+    }
+    else
+    {
+        bulkData.append(R"({"index":{"_index":")");
+        bulkData.append(index);
+        bulkData.append(R"("}})");
+        bulkData.append("\n");
+        bulkData.append(data);
+        bulkData.append("\n");
+    }
```

Comment:
This can be simplified
```suggestion
    bulkData.append(R"({"index":{"_index":")");
    bulkData.append(index);
    
    if (!id.empty()) {
        bulkData.append(R"(","_id":")");
        bulkData.append(id);
    }
    
    bulkData.append(R"("}})");
    bulkData.append("\n");
    bulkData.append(data);
    bulkData.append("\n");
```

Reply:
You are right. I'll commit the suggestion.